crates/compiler/test_gen/benches/quicksort.rs

```diff
@@ -0,0 +1,158 @@
+#[path = "../src/helpers/mod.rs"]
+mod helpers;
+
+// defines roc_alloc and friends
+pub use helpers::platform_functions::*;
+
+use bumpalo::Bump;
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use roc_gen_llvm::{llvm::build::LlvmBackendMode, run_roc::RocCallResult, run_roc_dylib};
+use roc_mono::ir::OptLevel;
+use roc_std::RocList;
+
+// results April 9, 2023
+//
+// > pure roc quicksort      time:   [106.97 us 107.27 us 107.63 us]
+// > roc zig quicksort       time:   [34.765 us 35.301 us 35.865 us]
+// > rust std sort           time:   [20.413 us 20.623 us 20.838 us]
```

Comment:
I suspect a big factor in this gap is this kind of pattern in the code
```
            let `#UserApp.33` : {U64, List I64} = CallByName `#UserApp.partition` `#UserApp.low` `#UserApp.high` `#UserApp.list`;
            let `#UserApp.partitionIndex` : U64 = StructAtIndex 0 `#UserApp.33`;
            let `#UserApp.partitioned` : List I64 = StructAtIndex 1 `#UserApp.33`;
            inc `#UserApp.partitioned`;
            dec `#UserApp.33`;
```

where we inc and then dec a list. This blocks optimization. But we should be able to cancel those operations agains each  other